CRM/Utils/Mail/EmailProcessor.php

```diff
@@ -118,7 +118,14 @@ private static function _process($civiMail, $dao, $is_create_activities) {
     // process fifty at a time, CRM-4002
     while ($mails = $store->fetchNext(MAIL_BATCH_SIZE)) {
       foreach ($mails as $key => $mail) {
-        $incomingMail = new CRM_Utils_Mail_IncomingMail($mail, (string) $dao->domain, (string) $dao->localpart);
+        try {
+          $incomingMail = new CRM_Utils_Mail_IncomingMail($mail, (string) $dao->domain, (string) $dao->localpart);
+        }
+        catch (CRM_Core_Exception $e) {
+          $store->markIgnored($key);
+          continue;
```

Comment:
I've put merge-ready just because right now it's a silent fail, so wondering if it should log something. I suppose people should be monitoring the ignored folder, but I admit I don't.